### DISCUSSION TITLE
Why Openthread require IPv4 forwarding ?

### DISCUSSION THREAD

**Discussion by caipiblack** [openthread/openthread#9783] [Q&A] [answered]
Hello,

Openthread require IPv6 and IPv4 forwarding, we have this information from various documents:

- https://openthread.io/guides/border-router/docker/run?hl=fr
- https://github.com/openthread/ot-br-posix/blob/main/script/_ipforward

I understand why Openthread require IPv6 forwarding:
- Because the Linux Device (for exemple: raspberry pi) needs to act as a router to permit the forwarding of messages from eth0 <> wpan0 interfaces so that we can communicate with Thread IPv6 network from outside.

But why is it necessary for IPv4 as Thread does not use IPv4 ?

(And I think, to access the border router web interface with IPv4, we don't need the forwarding)

**Comment by jwhui** [comment, marked as answer]
While Thread devices only support IPv6, they can communicate with external IPv4 end points via Thread Border Routers using NAT64. The NAT64 function requires IPv4 forwarding on the Thread Border Router.
